### PR TITLE
Pass certificates explicitly (without adapters)

### PR DESCRIPTION
**Pull request by thenno**
For https://github.com/datalens-tech/datalens-backend/issues/233

1. add CA_FILE_PATH to settings with an old default value
2. read ca file on app's startup
3. pass it to US and AIOHttp*-based clients (e.g. file uploader)
4. fix some unclosed sessions
5. add special `get_root_certificates*` functions for tests (this functions can read PATH from env)

### CODE REVIEW

File: lib/dl_api_commons/dl_api_commons/aiohttp/aiohttp_client.py [hunk 1]
```diff
@@ -112,26 +106,24 @@ class BIAioHTTPClient:
 
     retrier: BaseRetrier = attr.ib(factory=NoRetriesRetrier)
 
-    ssl_context: Optional[ssl.SSLContext] = attr.ib(factory=default_ssl_context)
-    _session: Optional[aiohttp.ClientSession] = attr.ib()
-    close_session_on_exit: Optional[bool] = attr.ib(default=None)
+    _ca_data: bytes = attr.ib()
```
Comment (ovsds) [inline lib/dl_api_commons/dl_api_commons/aiohttp/aiohttp_client.py:109]: why not to store str at the first place?

File: lib/dl_core/dl_core/united_storage_client_aio.py [hunk 2]
```diff
@@ -130,14 +129,17 @@ def __init__(
...
 
-        ssl_context = ssl.create_default_context(cafile=get_root_certificates_path())
-
-        self._session = aiohttp.ClientSession(
+        self._bi_http_client = BIAioHTTPClient(
+            base_url="/".join([self.host, self.prefix]),
+            retrier=PredefinedIntervalsRetrier(
+                retry_intervals=self._retry_intervals,
+                retry_codes=self._retry_codes,
+                retry_methods={"GET", "POST", "PUT", "DELETE"},  # TODO: really retry all of them?..
```
Comment (ovsds) [inline lib/dl_core/dl_core/united_storage_client_aio.py:137]: retriing non-indempotent methods sounds scary

File: lib/dl_api_commons/dl_api_commons/aiohttp/aiohttp_client.py (hunk 1 above)
Comment (thenno) [reply lib/dl_api_commons/dl_api_commons/aiohttp/aiohttp_client.py:109]: Some libraries (for example grpc) wants ca_data as bytes (`grpc.ssl_channel_credentials`).

Thus using bytes is a clearer way for all cases (bytes from disk -> bytes in python -> bytes to grpc | decoded string to aiohttp | etc)

File: lib/dl_core/dl_core/united_storage_client_aio.py (hunk 2 above)
Comment (thenno) [reply lib/dl_core/dl_core/united_storage_client_aio.py:137]: yes, it's kinda strange
https://github.com/datalens-tech/datalens-backend/issues/249

I was afraid to change it without discussion

File: lib/dl_core/dl_core_tests/conftest.py [hunk 3]
```diff
@@ -1 +1,11 @@
+import pytest
+
+from dl_testing.utils import get_root_certificates
+
+
 pytest_plugins = ("aiohttp.pytest_plugin",)  # and it, in turn, includes 'pytest_asyncio.plugin'
+
+
+@pytest.fixture(scope="function")
```
[the remark this answers is not in the document]
Comment (thenno) [reply lib/dl_core/dl_core_tests/conftest.py:9]: I wanted to set "function" here based on an existed code:
# chiffa @ chiffa-osx in ~/Code/datalens-backend on git:7ae48ce o [18:27:19]
$ grep -G --color=auto -R --exclude-dir={.git,.idea,.tox,.venv,.tools_venv,.ruff_cache,.mypy_cache,\*proto_stubs} "fixture(scope=\"function\")" | wc -l
     186

# chiffa @ chiffa-osx in ~/Code/datalens-backend on git:7ae48ce o [18:27:22]
$ grep -G --color=auto -R --exclude-dir={.git,.idea,.tox,.venv,.tools_venv,.ruff_cache,.mypy_cache,\*proto_stubs} "fixture()" | wc -l
       4

184 fixtures have explicitly set a "function" scope vs 4 without it.

**Review by ovsds — APPROVED** [review]

**Pull request merged**